sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasStateServer.scala

```diff
@@ -389,6 +520,63 @@ class TransformWithStateInPandasStateServer(
         } else {
           sendResponse(1, s"List state $stateName already exists")
         }
+        case StateVariableType.MapState => if (!mapStates.contains(stateName)) {
+          val valueSchema = StructType.fromString(mapStateValueSchemaString)
+          val valueExpressionEncoder = ExpressionEncoder(valueSchema).resolveAndBind()
+          val state = if (ttlDurationMs.isEmpty) {
+            statefulProcessorHandle.getMapState[Row, Row](stateName,
+              Encoders.row(schema), Encoders.row(valueSchema))
+          } else {
+            statefulProcessorHandle.getMapState[Row, Row](stateName, Encoders.row(schema),
+              Encoders.row(valueSchema), TTLConfig(Duration.ofMillis(ttlDurationMs.get)))
+          }
+          mapStates.put(stateName,
+            MapStateInfo(state, schema, valueSchema, expressionEncoder.createDeserializer(),
+              expressionEncoder.createSerializer(), valueExpressionEncoder.createDeserializer(),
+              valueExpressionEncoder.createSerializer()))
+          sendResponse(0)
+        } else {
+          sendResponse(1, s"Map state $stateName already exists")
+        }
+    }
+  }
+
+  private def sendIteratorAsArrowBatches[T](
```

Comment:
I see this is also added from #47878 but different implementation. Do you plan to rebase this PR to apply the new implementation to timer as well once #47878 is merged?

Reply:
yeah, if the timer PR is merged first, I can rebase and apply the update. Otherwise let's just keep the current implementation, we can do the rebase in the timer PR.